compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangImportPackage.java

```diff
@@ -130,8 +130,8 @@ public String getQualifiedPackageName() {
             versionStr = " version " + versionStr;
         }
 
-        String aliasStr = (this.version.value != null) ? this.version.value : "";
-        if (!aliasStr.isEmpty()) {
+        String aliasStr = (this.alias.value != null && !this.alias.value.equals(pkgName)) ? this.alias.value : "";
+        if (!aliasStr.isEmpty() & !aliasStr.equals(pkgName)) {
```

Comment:
```suggestion
        if (!aliasStr.isEmpty() && !aliasStr.equals(pkgName)) {
```

Reply:
We can combine this if with the above ternary.

Reply:
Fixed in 5d3fa6e.